Cql/Cql.Operators/CqlOperators.IntervalOperators.cs

```diff
@@ -364,10 +364,14 @@ internal partial class CqlOperators
                 // In other words, adjacent intervals within a sorted list are merged if they either overlap or meet.
                 if ((meets(x, y, precision) ?? false) || (OverlapsHelper(x, y, precision, toClosed) ?? false))
                 {
-                    var lowClosed = x.lowClosed;
-                    var highClosed = y.highClosed;
-
-                    return new CqlInterval<T?>(x.low, y.high, x.lowClosed, y.highClosed);
+                    if  (IntervalIncludesInterval(x, y, precision!) ?? false)

```

Comment:
Wouldn't we need to check for `IntervalIncludesInveral(y,x, precision1)` too?  In other words, does the same bug not exist when y properly includes x?

Reply:
No, because the intervals are first sorted by their low value and added to the queue:

`var queue = ListSortBy(intervals, i => i == null ? null! : i.low!, ListSortDirection.Ascending)?.ToList();`

**This means that interval x will always have a low value that is less than or equal to interval y low value**. In the event that the low values are the same, and the y interval properly includes the x interval, the outcome will be correct when we return:

`return new CqlInterval<T?>(x.low, y.high, x.lowClosed, y.highClosed);`

Example: 
```
interval1 = [20,29]
interval2 = [20,28]
```

After sorting the queue will be either
`queue = { [20, 28], [20, 29]}`
or 
`queue = { [20, 29], [20, 28]}`

In the first case, interval y properly include interval x and will return [x.low, y.high] ( [20,29] )

In the second case interval x properly includes includes y which triggers the conditional and return x ( [20,29] )

Reply:
Perfect, just double checking, thanks!